connection.impl/src/main/java/com/ericsson/bss/cassandra/ecchronos/connection/impl/builders/DistributedNativeBuilder.java

```diff
@@ -244,10 +244,15 @@ public final DistributedNativeConnectionProviderImpl build()
         LOG.info("Requesting Nodes List");
         List<Node> nodesList = createNodesList(session);
         LOG.info("Nodes list was created with success");
-        return new DistributedNativeConnectionProviderImpl(session, nodesList);
+        return new DistributedNativeConnectionProviderImpl(session, nodesList, this);
```

Comment:
Not sure this is the correct way of doing this. 
I need code that is part of the builder so when nodes are added or deleted, or currently pass the builder in and re use it. 
But maybe this should more of a refactor to take the required code out of the builder